osf/utils/fields.py

```diff
@@ -12,7 +12,7 @@ class LowercaseCharField(models.CharField):
     def get_prep_value(self, value):
         value = super(models.CharField, self).get_prep_value(value)
         if value is not None:
-            value = value.lower()
+            value = str(value).lower()
```

Comment:
Does this run before validation happens? If so, then we run the risk of putting bad data in the database. For example, if an `int` is passed to this field, it will silently be cast to a `str`.